yt_dlp/YoutubeDL.py

```diff
@@ -2163,6 +2166,18 @@ def can_merge():
             merger = FFmpegMergerPP(self)
             return merger.available and merger.can_merge()
 
+        def evaluate_formats(format_spec):
+            formats = info_dict.get('formats') or []
+            return list(self.build_format_selector(format_spec)({
+                'formats': formats,
+                'has_merged_format': any('none' not in (f.get('acodec'), f.get('vcodec')) for f in formats),
+                'incomplete_formats': (all(f.get('vcodec') == 'none' for f in formats)  # No formats with video
+                                       or all(f.get('acodec') == 'none' for f in formats)),  # OR, No formats with audio
+            }))
+        if not can_merge() and evaluate_formats('best/bestvideo+bestaudio') != evaluate_formats('bestvideo*+bestaudio/best'):
```

Comment:
This may cause some unnecessary checks when `--check-formats` is used without ffmpeg. But ig that's fine

Reply:
Yeah, I couldn't think of a nice way to prevent this. With the `__working` flag we can at least prevent the same format getting checked multiple times.
I think no ffmpeg, uses `--check-formats`, and doesn't use `-f` should be sufficiently rare that this is fine.